src/components/touchables/splitStyleProp.ts

```diff
@@ -0,0 +1,61 @@
+import { FlexStyle, StyleProp, StyleSheet } from 'react-native';
+
+const OUTER_PROPS: { [key in keyof FlexStyle]?: true } = {
```

Comment:
I'm not sure if these are really all the style props that need to be applied to the outer view or if there are more (e.g. `backfaceVisibility`, `elevation`, `transform` and all of `ShadowStyleIOS`).

It would probably be best to add complex layout examples to test this, as you mentioned.